### PR TITLE
Support input from stdin

### PR DESCRIPTION
**Pull request by danyill**
This is perhaps not quite right but might be close to being right. It is at the limit of my javascript ability. I think the cleanup of temporary html/adoc files needs to be better but it does seem to work :wink:

related to #155

### CODE REVIEW

**Comment by Mogztter** [comment]
Oh thanks!
That's quite good but I don't want to write a file if the content is read from stdin. I will need to refactor the code because the converter wants a file.

I also think that the Asciidoctor.js CLI should export a function to read from stdin:

```js
const data = await Invoker.readFromStdin()
```

But I will definitely build upon your work :+1:

**Comment by Mogztter** [comment]
@danyill if you want to give it a try I can describe in detail what's (in my opinion) needs to be done?

**Comment by danyill** [comment]
Yes please. I have some time and would like to try

**Comment by Mogztter** [comment]
@danyill Thanks for stepping up, I've edited my comment, please let me know if it's unclear :wink:

File: lib/converter.js [hunk 1]
```diff
@@ -55,9 +55,10 @@ async function convert (processor, inputFile, options, timings, watch, preview)
   }
   const instanceOptions = Object.assign({}, options, { to_file: tempFile })
   let doc
-
+  let timer
+  
   if (timings) {
-    const timer = processor.Timings.$new()
+    timer = processor.Timings.$new()
```
Comment (Mogztter) [inline lib/converter.js]: you can use `processor.Timings.create()` here, it's an alias.

**Comment by danyill** [comment]
@Mogztter thank you for your guidance :clap:, I think I understand and the result is cleaner and more elegant. I'd be happy to take further comments and make any amendments.

Now :beach_umbrella: and :sun_with_face:

File: lib/cli.js [hunk 2]
```diff
@@ -9,11 +9,14 @@ class ProcessorEmitter extends EventEmitter {}
...
 const converter = require('./converter.js')
+// we look forward to the CLI having Invoker.readFromStdin()
+// https://github.com/Mogztter/asciidoctor-pdf.js/pull/158#issuecomment-577788688
+const stdin = require('./stdin')
 
 async function convertFiles (files, argv, options, verbose, preview) {
   for (const file of files) {
     if (verbose) {
-      console.log(`converting file ${file}`)
+      console.log(`converting file ${file.contents ? '-' : file.path}`)
```
Comment (Mogztter) [inline lib/cli.js]: It's working but in theory we could have a file with a contents and a path, so I would write:

```suggestion
      console.log(`converting file ${file.path || '-'}`)
```

**Comment by Mogztter** [comment]
It looks great, thanks @danyill :+1: 
I hope you had a nice day at the :beach_umbrella:

**Pull request merged**